dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/TaskInstance.java

```diff
@@ -298,6 +305,14 @@ public void setState(ExecutionStatus state) {
         this.state = state;
     }
 
+    public Date getFirstSubmitTime() {
+        return firstSubmitTime;
```

Comment:
If a task can be retried after failure and has been delayed enough time, it will no longer enter the state of delayed execution. This attribute is used to achieve this feature.